plugins/redis/redisasync.cpp

```diff
@@ -0,0 +1,859 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2014 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+#include "hiredis/adapters/libev.h"
+
+#include "platform.h"
+#include "eclrtl.hpp"
+#include "jstring.hpp"
+#include "jsem.hpp"
+#include "jsocket.hpp"
+#include "jthread.hpp"
+#include "jmutex.hpp"
+#include "redisplugin.hpp"
+#include "redissync.hpp"
+#include "redisasync.hpp"
+
+#include "hiredis/async.h"
+
+namespace Lock
```

Comment:
You should not be using namespaces to differentiate parts of the code (and then reusing names within those namespaces) - it's VERY confusing!

All the code in the redis plugin should be in one namespace, and you should use different classnames for the different types.